Backoffice Login, Dashboard, Accounts, Transactions page

hafidnrzs opened the pull request.

hafidnrzs committed:
feat: backoffice login page

hafidnrzs committed:
feat: Create a separate "InputForm" molecule component

- Develop the "InputForm" molecule component
- Remove empty modal-gacha component stories and test files that prevent Storybook from running

hafidnrzs committed:
feat: Update InputForm component

- Perbarui docs untuk component Input dan InputForm
- Perbarui component InputForm agar bekerja dengan helper text, error, dan label/input

hafidnrzs committed:
feat: Update InputForm to handle different sizes

- Tambah spec test pada komponen InputForm
- Tambah autodocs pada komponen Input

hafidnrzs committed:
feat: Update InputForm, Input, and auth page

- Perbaiki page login ke halaman utama backoffice
- Perbaiki component Input dan InputForm

hafidnrzs committed:
feat: Draft dashboard AI-generated

hafidnrzs committed:
feat: Update sidebar

- Perbaiki tampilan sidebar sesuai desain

hafidnrzs committed:
feat: summary section in dashboard page

hafidnrzs committed:
feat: gacha section in dashboard

hafidnrzs committed:
feat: Fix gacha image preview

- Menggunakan pendekatan simpan gambar clipped langsung daripada mengatur secara dinamis menggunakan CSS object-position

hafidnrzs committed:
feat: Another page template

- Template untuk page Data Akun dan Validasi Transaksi

hafidnrzs committed:
feat: Divider on top of logout button sidebar

hafidnrzs committed:
feat: Draft Data Akun page

- Vibes coding untuk siapkan base dari halaman Data Akun

hafidnrzs committed:
feat: Refactor datatable and pagination component and add storybook

- Refactor nama component dari datatable dan pagination mengikuti codebase
- Buat storybook untuk preview dari masing-masing komponen

hafidnrzs committed:
feat: Styling data table component

hafidnrzs committed:
feat: Styling pagination

hafidnrzs committed:
feat: Page Data Akun

hafidnrzs committed:
feat: Draft transaction page

hafidnrzs committed:
feat: Change data table component dynamically

- Ubah komponen DataTable agar bisa render menyesuaikan jumlah kolom dan baris pada data

hafidnrzs committed:
feat: Adjustment

- Sesuaikan color palette success, info, danger, warning sesuai dengan design system Figma
- Sesuai background pagination
- Fix width form login tidak maksimal

hafidnrzs commented:
**Changes:**

- Backoffice: Tambah page Login, Dashboard, Data Akun, Validasi Transaksi

![backoffice-login](https://github.com/user-attachments/assets/bbc51ce1-02f8-4886-9336-67c5cdde24e5)
![backoffice-dashboard](https://github.com/user-attachments/assets/003692c3-cbc6-4b0c-ac63-45591f727647)
![backoffice-data-akun](https://github.com/user-attachments/assets/2e598ccb-0d53-4d87-904b-316fbcd34e4f)
![backoffice-validasi-transaksi](https://github.com/user-attachments/assets/57702272-b796-457b-92e3-87d7c05d4b1b)

- Edit color palette di file `index.css` pada gacha, dimentorin, dan backoffice menyesuaikan Design System pada Figma
- Gabungkan component PasswordInput ke Input
- Buat component InputForm yang menggabungkan Input dengan label dan helper text/error text
- Buat component Pagination
- Buat component DataTable untuk menampilkan tabel dan reusable menyesuaikan kolom dan data. Contoh penerapan pada Backoffice halaman Data Akun dan Validasi Transaksi

![image](https://github.com/user-attachments/assets/25d68e1d-6e41-44f7-b018-3841bffdeb5a)
![image](https://github.com/user-attachments/assets/2337b99c-f239-4344-b131-8a856df95c46)
![image](https://github.com/user-attachments/assets/17a8898b-2301-49e2-ae63-83ddc16ce7da)

- Buat component BackofficeSidebar
- Hapus storybook dan test file ModalGacha karena menyebabkan error saat menjalankan storybook

hafidnrzs committed:
feat: fix datatable storybook

hafidnrzs committed:
fix: Sidebar stick to left. Fit content for different desktop viewports

hafidnrzs committed:
fix: Sidebar stick to left. Fit content for different desktop viewports

hafidnrzs committed:
Merge branch into backoffice/auth

hafidnrzs commented:
Update Backoffice Sidebar agar menempel ke sebelah kiri

![image](https://github.com/user-attachments/assets/d09860e7-1da9-4d9d-bd10-a1aa799536bd)

maulanasdqn commented on libs/ui/src/organisms/datatable/datatable.tsx:
pake tanstack react table 8 bang

hafidnrzs committed:
feat: Add TanStack Table v8 and implement it to Datatable component

hafidnrzs committed:
feat: Simple implementation TanStack table

Coba implementasikan versi paling sederhana TanStack Table v8 langsung ke page Data Akun

hafidnrzs committed:
feat: Implement pagination

- Terapkan pagination pada page Data Akun

hafidnrzs committed:
feat: Working TanStack pagination

- Menerapkan TanStack Table dan Pagination

hafidnrzs committed:
feat: separate DataTable component

hafidnrzs committed:
feat: Combine Pagination component into DataTable

hafidnrzs committed:
feat: Implement DataTable to backoffice pages

- Halaman Data Akun
- Halaman Validasi Transaksi
